scripts/registry-validate-source.py

```
@@ -76,22 +89,70 @@ def _validate_integrations(doc: dict) -> None:
+            if not any(field in integration for field in ("source", "linux_systemd_user", "macos_launch_agent", "windows_service")):
+                raise ValidationError(f"{prefix} must declare at least one service source")
+            for field in ("source", "linux_systemd_user", "macos_launch_agent", "windows_service"):
+                if field in integration:
+                    source = _expect_non_empty_str(integration, field, prefix)
+                    _validate_integration_source(source, f"{prefix}.{field}")
         if key in seen:
             raise ValidationError(f"duplicate integration declaration {key}")
         seen.add(key)
 
 
+def _validate_shell_init(doc: dict) -> None:
```

Ruff (0.15.12)

[warning] Too many branches (15 > 12)

(PLR0912)